urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/cache/InstantMsgCache.java

```diff
@@ -0,0 +1,458 @@
+				while (iter.hasNext()) {
+					Map.Entry<String, DynamicInfoSerializable> entry = iter.next();
+
+					DynamicInfoSerializable dis = entry.getValue();
+
+					// 取出信息
+					String nodeName = dis.getNodeName();
+					String updateTime = dis.getUpdateTime();
+					float oneMinsProcs = dis.getOneMinsProcs();
+					float fiveMinsProcs = dis.getFiveMinsProcs();
+					float fifteenMinsProcs = dis.getFifteenMinsProcs();
+					float userTime = dis.getUserTime();
+					float niceTime = dis.getNiceTime();
+					float systemTime = dis.getSystemTime();
+					float iowaitTime = dis.getIowaitTime();
+					float idleTime = dis.getIdleTime();
+					String allUserTime = dis.getAllUserTime();
+					String allNiceTime = dis.getAllNiceTime();
+					String allSystemTime = dis.getAllSystemTime();
+					String allIowaitTime = dis.getAllIowaitTime();
+					String allIdleTime = dis.getAllIdleTime();
+					int totalMemory = dis.getTotalMemory();
+					int usedMemory = dis.getUsedMemory();
+					int idleMemory = dis.getIdleMemory();
+					int swapSize = dis.getSwapSize();
+					int usedSwap = dis.getUsedSwap();
+					int idleSwap = dis.getIdleSwap();
+					int ioTs = dis.getIoTs();
+					int readSpeed = dis.getReadSpeed();
+					int readKb = dis.getReadKB();
+					int writeSpeed = dis.getWriteSpeed();
+					int writeKb = dis.getWriteKB();
+					double inReceivesPs = dis.getInReceivesPs();
+					double inDeliversPs = dis.getInDeliversPs();
+					double outRequestsPs = dis.getOutRequestsPs();
+					double inSegsPs = dis.getInSegsPs();
+					double outSegsPs = dis.getOutSegsPs();
+					double retransSegsPs = dis.getRetransSegsPs();
+					double inDatagramsPs = dis.getInDatagramsPs();
+					double outDatagramsPs = dis.getOutDatagramsPs();
+					stmt.executeUpdate(SQLConstants.updateComputeNodeStatusMySQL(nodeName, updateTime, oneMinsProcs, fiveMinsProcs, fifteenMinsProcs, userTime, niceTime, systemTime, iowaitTime, idleTime, allUserTime, allNiceTime, allSystemTime, allIowaitTime, allIdleTime, totalMemory, usedMemory, idleMemory, swapSize, usedSwap, idleSwap, ioTs, readSpeed, readKb, writeSpeed, writeKb, inReceivesPs, inDeliversPs, outRequestsPs, inSegsPs, outSegsPs, retransSegsPs, inDatagramsPs, outDatagramsPs));
+					conn.commit();
+					// MaintainInfo.computeNodeStatusCount++;
+				}
+			}
+		}
+
+		private boolean dbTypeIsOracle() {
+			if (dbType.equalsIgnoreCase("Oracle")) {
+				return true;
+			} else {
+				return false;
+			}
+		}
+
+		private boolean dbTypeIsMySQL() {
+			if (dbType.equalsIgnoreCase("MySQL")) {
+				return true;
+			} else {
+				return false;
+			}
+		}
+
+		private void initialOracleDB() throws SQLException {
+			conn = jdbcCommon.getOrclConn();
+			if (conn == null) {
+				LOG.error("Can't connect to Oracle-->" + jdbcCommon.getIpAddress());
+			} else {
+				LOG.info("Sucessfully connect to Oracle-->" + jdbcCommon.getIpAddress());
+				conn.setAutoCommit(false);
+				stmt = conn.createStatement();
+				setStoreData(true);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 282:289, 294:301.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.